### PR TITLE
refactor(zk-kit-poseidon): remove old circomlib dependencies in crypto

### PR DESCRIPTION
**Pull request by ctrlc03**
# Description

This PR removes an old import (ciromlib) from the crypto library and instead imports the same functions from [zk-kit](https://github.com/privacy-scaling-explorations/zk-kit). zk-kit provides lightweight TypeScript libraries for zk developers, allowing MACI's packages to be slimmer and more up to date.

## Additional Notes

This is the first step into removing an old [dependency](https://github.com/weijiekoh/circomlib), currently we still use it inside the circom circuits, and will be tackled in another PR.

@cedoor MACI 🤝 zk-kit 

## Related issue(s)

Re: #772 

## Additional Info

Let's merge #944 first 

## Confirmation

- [x] I have read and understand MACI's [contributor guidelines](https://github.com/privacy-scaling-explorations/maci/blob/dev/CONTRIBUTING.md) and [code of conduct](https://github.com/privacy-scaling-explorations/maci/blob/dev/CODE_OF_CONDUCT.md).
- [x] I have read and understand MACI's [GitHub processes](https://github.com/privacy-scaling-explorations/maci/discussions/847).
- [x] I have read and understand MACI's [testing guide](https://maci.pse.dev/docs/testing).

### CODE REVIEW

File: crypto/ts/types.ts [hunk 1]
```diff
@@ -1,10 +1,24 @@
-export type SnarkBigInt = bigint;
-export type PrivKey = bigint;
-export type PubKey = bigint[];
-export type Point = bigint[];
-export type EcdhSharedKey = bigint[];
-export type Plaintext = bigint[];
-export type Ciphertext = bigint[];
+// we define a bignumber as either a bigint or a string
+// which is what we use the most in MACI
+export type BigNumber = bigint | string;
```
Comment (0xmad) [inline crypto/ts/types.ts:3]: I think this type can have conflicts if it's using with BigNumber from ethers of from bignumber.js. Maybe it makes sense to reuse BigNumberish type from ethers?

**Review by 0xmad — APPROVED** [review]
@ctrlc03 thanks, just one comment/question.

File: crypto/ts/types.ts (hunk 1 above)
Comment (ctrlc03) [reply crypto/ts/types.ts:3]: Good shout, I can use that instead

Comment (ctrlc03) [reply crypto/ts/types.ts]: What do you think about simply changing the type name? so we don't import from ethers and make further changes to the code

Comment (0xmad) [reply crypto/ts/types.ts]: I think it works as well.

**Pull request merged**